x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/NumberUtils.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.profiling;
+
+final class NumberUtils {
+    private NumberUtils() {
+        // no instances intended
+    }
+
+    /**
+     * Converts a positive double number to a string.
+     *
+     * @param value The double value.
+     * @return The corresponding string representation truncated to four fractional digits.
+     */
+    public static String doubleToString(double value) {
+        StringBuilder sb = new StringBuilder();
```

Comment:
In the `exclusive` arrays of the flamegraph, we have lots of `0` values, like
```
  "AnnualCO2TonsExclusive": [
    0,
    0,
    0,
    0,
...
```
Doesn't this code send `0.0000` instead of `0`?
I'd say that values < 0.0001 can be sent as just `0`.
Does it make sense to add that check before calling `new StringBuilder()`?

Reply:
Great idea! That should reduce the size of the response even further. I've implemented this in https://github.com/elastic/elasticsearch/pull/105779/commits/a23e0edf9adf779849fe298bd209f6b2cf64cfc3.